app/views/layouts/_navbar.html.haml

```diff
@@ -48,6 +48,3 @@
             =link_to events_path do
               %i.fa.fa-dashboard
               %span Dashboard
-
-- if params[:controller].include?("staff") && user_signed_in?
-  = render partial: "layouts/nav/staff_subnav"
```

Comment:
Moving the render partial calls to each page works, I'd prefer going back to using our navbar partial since that's the logical place I'd look for it.

The original check doesn't communicate why the staff_subnav should or should not show up, regardless of if it worked.
if params[:controller].include?("staff") && user_signed_in?

What if we tried something that reads better and is more flexible?

So go back to your single render partial in layouts/_navbar:
= render partial: "nav/staff_subnav" if display_staff_nav?

Now it's conditional on with a post if check.  It also reads well.

Then in Staff::ApplicationController you can add two private methods:
enable_staff_subnav
display_staff_subnav?

enable_staff_subnav would set an instance variable like @display_staff_subnav = true

display_staff_subnav? would just return the value of that ivar.  You'll want to set it as a helper_method so that the view layer can access it.

We scope them by the Staff::ApplicationController since it only needs to be used by staff.

Finally, in each staff controller that should have the staff subnav add this:
before_action :enable_staff_subnav

Though it does add more code in some places it's much nicer to read and adding/removing the subnav from any event controller would be a one-liner that you could easily notice.  Try that approach and see how it feels.